Assets/Scripts/KatamariMovement.cs

```diff
@@ -342,22 +309,131 @@ void PowerBar()
+                powerBar.SetActive(false);
+                this.GetComponent<LineRenderer>().enabled = false;
+                resourceManager.tryToHit();
+                forceMode = ForceMode.Force;
+                
+            }
+        }
+
+        if (myStateMachine == stateMachine.toGolfMode) {
+            Time.timeScale += (1f / slowdownLength) * Time.unscaledDeltaTime;
+            Time.timeScale = Mathf.Clamp(Time.timeScale, slowdownFactor, 1f);
+            Time.fixedDeltaTime = Time.timeScale * 0.02f;
 
-    
-    
+            if (Time.timeScale == 1)
+            {
+                power = 0;
+                powerBar.SetActive(true);
+                this.GetComponent<LineRenderer>().enabled = true;
+                go_up = true;
+                forceMode = ForceMode.Impulse;
+                myStateMachine = stateMachine.golfMode;
 
+            }
 
+        }
+
+        //NORMAL SPEED\\
+        if (myStateMachine == stateMachine.normalSpeed)
+        {
+            // Transition to Slow Down\\
+            if (isFlyMovementDeadTime && hitInput > 0.5) {
+                myStateMachine = stateMachine.slowDown;
+            }
+            //Or toGolfMOde\\
+            if (resourceManager.can_hit()) {
+                myStateMachine = stateMachine.toGolfMode;
+            }
+
+        } else if (myStateMachine == stateMachine.slowDown) {
+            Time.timeScale -= (1f / slowdownLength) * Time.unscaledDeltaTime;
+            Time.timeScale = Mathf.Clamp(Time.timeScale, slowdownFactor, 1f);
+            Time.fixedDeltaTime = Time.timeScale * 0.02f;
+            flyStrength =0;
+
+            //Transition to slowMotion\\
+            if (Time.timeScale == slowdownFactor) {
+                myStateMachine = stateMachine.slowMotion;
+                flyStrength = standardStrength * (1 / slowdownFactor);
+            }
+            //Transition to Speed up if Played lets go of slow motion movement mid slowdown\\
+            if (hitInput <= 0.5){
+                myStateMachine = stateMachine.speedUp;
+            } //Or to toGolfMode\\
+            if (resourceManager.can_hit()){
+                myStateMachine = stateMachine.toGolfMode;
+            }
+
+        } else if (myStateMachine == stateMachine.slowMotion) {
+            if (hitInput < 0.5) {
+                ///Transition to Speed Up\\\
+                myStateMachine = stateMachine.speedUp;
+            }
+            if (resourceManager.can_hit()){
+                myStateMachine = stateMachine.toGolfMode;
+            }
+            //SPEED UP\\
+        } else if (myStateMachine == stateMachine.speedUp) {
+            Time.timeScale += (1f / slowdownLength) * Time.unscaledDeltaTime;
+            Time.timeScale = Mathf.Clamp(Time.timeScale, slowdownFactor, 1f);
+            Time.fixedDeltaTime = Time.timeScale * 0.02f;
+            flyStrength = 0;
+
+            rb.velocity *= .98f; // maybe a better more dynamic way to do this but this works for now
```

Comment:
Does the time scaling not feel right velocity-wise without this?

Reply:
its necessary